### DISCUSSION TITLE
Acrylic/Blur distorts small background details

### DISCUSSION THREAD

**Discussion by Undisputed00x** [ALTaleX531/OpenGlass#59] [Q&A] [answered]
Like the title says when using Blur or Acrylic the background has very poor quality on small details like text. When applied less blur then also those visual distortions are fewer but still existent
Probably im doing something wrong that's why i need help 😅

OpenGlass v1.3
Win11 24H2
Using MFE to extends effects entire window
using EBM to have composited texts

<details>
<summary>

### Video example
</summary>
<p>

https://github.com/user-attachments/assets/8985512e-d4b8-4861-b7d3-d71cf7df7939

</p>
</details>

**Comment by ALTaleX531** [comment]
Be smart and don't set GlassOpacity to 0. I'd recommend at least 40, or you'll continue to suffer from this.

Edit:
In addition to this, set GlassAdditionalPreScaleAmount to 100.

**Reply by Undisputed00x** [reply]
> Edit: In addition to this, set GlassAdditionalPreScaleAmount to 100.

Before posting it was 100



> Be smart and don't set GlassOpacity to 0. I'd recommend at least 40, or you'll continue to suffer from this.

Doesn't fix it and changes color of the frame (i don't even tried to learn how glass attributes work lol)


<details>
<summary>

### Video

</summary>
<p>

https://github.com/user-attachments/assets/a49415dc-7d20-408a-8d96-70ecb9f177ce

</p>
</details>


Anyway i can live with it until hopefully DWMBG gets an update (if ever will).
Grabbing the chance to tell you that shutting down openglass it crashes my entire system (im latest 24H2)

**Reply by ALTaleX531** [reply, marked as answer]
https://github.com/ALTaleX531/OpenGlass/blob/6002f01e6f908f566f8fe5087ce4dc8303fb38e3/OpenGlass/BlurBackdrop.hpp#L33
I see what's going on, you'll need to manually edit the code here because the quality of the blur is hardcoded to be the lowest because of performance consideration.

I can't reproduce the crash here, tested on build 26100.1742.

**Reply by Undisputed00x** [reply]
ok added `D2D1_GAUSSIANBLUR_OPTIMIZATION_QUALITY` to `D2D1_GAUSSIANBLUR_OPTIMIZATION`. Launched the new compiled dll but still no change to the issue.

<details>
<summary>

### Video

</summary>
<p>

https://github.com/user-attachments/assets/af8fc6f1-d476-4576-bed1-5803c8478ce8

</p>
</details>

**Reply by ALTaleX531** [reply]
I see that you just modified the default parameter of the function without modifying the actual call parameter of the function, I suggest replacing `BoxValue(optimizationMode)` with `BoxValue(D2D1_GAUSSIANBLUR_OPTIMIZATION_QUALITY)` in the video above.

**Reply by Undisputed00x** [reply]
> I see that you just modified the default parameter of the function without modifying the actual call parameter of the function,

How to tell someone a C++ noobie 😅 , time to rewatch those indian lectures on YT

> https://github.com/ALTaleX531/OpenGlass/blob/6002f01e6f908f566f8fe5087ce4dc8303fb38e3/OpenGlass/BlurBackdrop.hpp#L33
> 
> I see what's going on, you'll need to manually edit the code here because the quality of the blur is hardcoded to be the lowest because of performance consideration.

Holy sh*t 🚀, Blur quality got distinguishable better which also fixed the visual background glitches. Thanks!
Could you add a reg key for that in a hotfix maybe ?

<details>
<summary>

### Video

</summary>
<p>

https://github.com/user-attachments/assets/5518cb53-29ee-4600-afe0-4072dfbbbf38

</p>
</details>

> I can't reproduce the crash here, tested on build 26100.1742.

Im on build 26100.2161, closed all other mods i was using and still shutting down OpenGlass freezes (doesn't crash it like BSOD) the entire system (only mouse is moving nothing else responds not even shutting down the system through the case button)

**Reply by ALTaleX531** [reply]
I probably won't provide the option to configure the blur quality, as removing the tint color wouldn't be a wise choice, this is not recommended. By the way, I upgraded to the same version as you and still can't reproduce the crash, I'm guessing you didn't disable all the mods, you can upload a dump file to prove it.

I will say, though, that you installed the optional update KB5044384, and if you really want OpenGlass to survive longer, don't update your system **too often**. I can't imagine Microsoft being able to get through the next version **without** messing anything up.

**Reply by Skyyblaze** [reply]
I do have the "OpenGlass freezes everything" glitch as well on 24H2, however only the one time when I tried to uninstall OpenGlass because I was testing things.

**Reply by froggz19** [reply]
Can also confirm that shutting down OpenGlass by running shutdown.bat freezes the entire system.

Windows 11 26100.2314
